pkgs/applications/audio/ctrlr/default.nix

```diff
@@ -0,0 +1,73 @@
+{ stdenv, fetchFromGitHub, libGL, unzip, libX11, libbfd, zlib, udev, alsaLib, freetype
+, libXrandr, libXinerama, libXext, libXcomposite, libXcursor, libiberty, zenity, which
+, makeDesktopItem }:
+
+let
+  desktopItem = makeDesktopItem rec {
+    name = "Ctrlr";
+    exec = name;
+    desktopName = name;
+    genericName = "Ctrlr MIDI panel";
+    categories = "Audio;AudioVideo;";
+  };
+in stdenv.mkDerivation rec {
+  pname = "ctrlr";
+  version = "5.3.201";
+
+  src = fetchFromGitHub {
+    owner = "RomanKubiak";
+    repo = "ctrlr";
+    # Scripts/git-revision.sh
+    rev = "179e457a5280566df65645b49bebf6b424c3f4ca";
+    sha256 = "0n4s896m2zpbhj13iz77nv2686shwgrwbsdk36f3v84gqz44925s";
+  };
+
+  patches = [
+    ./patch-libr.h
+    ./patch-LAudio.cpp
+  ];
+
+  buildInputs = [
+    libGL libX11 libbfd zlib udev alsaLib freetype libXrandr libXinerama libXext
+    libXcomposite libXcursor libiberty
+  ];
+  nativeBuildInputs = [ unzip ];
+
+  NIX_CFLAGS_COMPILE = [
+    "-DJUCE_USE_FLAC=0"
+    "-I../../../../Source/Misc/include/lo"
+  ];
+
+  makeFlags = [
+    "CONFIG=Release"
+  ];
+
+  buildPhase = ''
+    (cd Boost && unzip -q boost.zip)
+    substitute Source/Core/CtrlrRevision.template ./Source/Core/CtrlrRevision.h \
+      --replace "%REVISION%" "${version}" \
+      --replace "%REVISION_DATE%" ""
+    substituteInPlace Builds/Generated/Linux/Standalone/Makefile \
+      --replace "libbfd-\$(ARCH).a" "-lbfd" --replace "libiberty-\$(ARCH).a" "-liberty"
+    substituteInPlace Juce/modules/juce_gui_basics/native/juce_linux_FileChooser.cpp \
+      --replace '"which' '"${which}/bin/which' \
+      --replace '"zenity' '"${zenity}/bin/zenity'
+    (cd Builds/Generated/Linux/Standalone && make -j$NIX_BUILD_CORES $makeFlags)
```

Comment:
Please avoid subshells and use ``cd ././.`` and ``cd ..`` or ``pushd`` and ``popd``.